sympy/tensor/array/dense_ndim_array.py

```diff
@@ -59,6 +59,9 @@ def __getitem__(self, index):
             index = tuple([i for i in index] + \
                           [slice(None) for i in range(len(index), self.rank())])
 
+        if len(index) > self.rank():
+            raise ValueError('Dimension of index greater than rank of array')
```

Comment:
maybe this should be relocated to the superclass, in order not to repeat the code?